src/main/java/org/opentripplanner/inspector/ElevationEdgeRenderer.java

```diff
@@ -0,0 +1,105 @@
+package org.opentripplanner.inspector;
+
+import java.awt.Color;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+import org.opentripplanner.common.model.T2;
+import org.opentripplanner.inspector.EdgeVertexTileRenderer.EdgeVertexRenderer;
+import org.opentripplanner.inspector.EdgeVertexTileRenderer.EdgeVisualAttributes;
+import org.opentripplanner.inspector.EdgeVertexTileRenderer.VertexVisualAttributes;
+import org.opentripplanner.routing.edgetype.StreetEdge;
+import org.opentripplanner.routing.graph.Edge;
+import org.opentripplanner.routing.graph.Graph;
+import org.opentripplanner.routing.graph.Vertex;
+
+class ElevationEdgeRenderer implements EdgeVertexRenderer {
+
+    private final ScalarColorPalette colorPalette;
+
+    ElevationEdgeRenderer(Graph graph) {
+        if (graph.hasElevation) {
+            colorPalette = new DefaultScalarColorPalette(
+                    graph.minElevation,
+                    (graph.minElevation + graph.maxElevation) / 2,
+                    graph.maxElevation
+            );
+        } else {
+            colorPalette = new DefaultScalarColorPalette(0, 0, 0);
+        }
+    }
+
+    @Override
+    public boolean renderEdge(
+            Edge e, EdgeVisualAttributes attrs
+    ) {
+        return true;
+    }
+
+    @Override
+    public boolean renderVertex(
+            Vertex v, VertexVisualAttributes attrs
+    ) {
+        var elevation = findElevationForVertex(v);
+        if (elevation != null) {
+            attrs.color = colorPalette.getColor(elevation);
+            attrs.label = elevation.toString();
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    @Override
+    public boolean hasEdgeSegments(Edge edge) {
+        return true;
+    }
+
+    @Override
+    public Iterable<T2<Double, Color>> edgeSegments(Edge edge) {
+        if (edge instanceof StreetEdge) {
+            if (((StreetEdge) edge).hasElevationExtension()) {
```

Comment:
Java 17 has a pretty nice feature to avoid this cast: http://www.tutorialsforgeeks.org/java17/java17_pattern_matching_instanceof.htm